### ISSUE TITLE
feat(c++): raise error at runtime if arrow does not support ORC

### ISSUE THREAD

**Issue by acezen** [apache/incubator-graphar#546]
### Describe the enhancement requested
the follow-up issue of #545, try to raise error at runtime if arrow does not support ORC.

the ORC related code list below:

https://github.com/apache/incubator-graphar/blob/d00c3649d2fdee6ba0fbf612b473707688d60e3f/cpp/src/graphar/filesystem.cc#L20

https://github.com/apache/incubator-graphar/blob/d2e8196443bb62155fe55e47752d8e259f7c4ba7/cpp/src/graphar/filesystem.cc#L93

https://github.com/apache/incubator-graphar/blob/d00c3649d2fdee6ba0fbf612b473707688d60e3f/cpp/src/graphar/filesystem.cc#L236-L244

https://github.com/apache/incubator-graphar/blob/d00c3649d2fdee6ba0fbf612b473707688d60e3f/cpp/src/graphar/reader_util.cc#L20

https://github.com/apache/incubator-graphar/blob/d00c3649d2fdee6ba0fbf612b473707688d60e3f/cpp/test/test_arrow_chunk_writer.cc#L25

https://github.com/apache/incubator-graphar/blob/d2e8196443bb62155fe55e47752d8e259f7c4ba7/cpp/test/test_arrow_chunk_writer.cc#L115

https://github.com/apache/incubator-graphar/blob/d2e8196443bb62155fe55e47752d8e259f7c4ba7/cpp/test/test_builder.cc#L27

### Solution
Maybe add a macro definition in CMakeList to indicate `ARROW_ORC` support or not.

### Component(s)

C++

**Comment by yecol** [comment]
Hi @acezen, 

Just try to understand more about this issue.
After #552 merged, the CMakeList will prevent building if the arrow does not build with ORC. 
So I just wander, 
1) in what situation the built GraphAr will not have ORC in runtime?
2) or GraphAr can be run without ORC support BY DESIGN? in this case, just disabled the ORC related functionalities if ORC is not found?

**Comment by acezen** [comment]
> Hi @acezen, 
> 
> Just try to understand more about this issue.
> After #552 merged, the CMakeList will prevent building if the arrow does not build with ORC. 
> So I just wander, 
> 1) in what situation the built GraphAr will not have ORC in runtime?
> 2) or GraphAr can be run without ORC support BY DESIGN? in this case, just disabled the ORC related functionalities if ORC is not found? 

1. GraphAr support the ORC through arrow, so if arrow without ORC support, GraphAr can not support ORC read/write in runtime.
2. There may have situations that user only use parquet/csv other than ORC as payload file format, it may not need ORC support.

I suggest we can turn the without ORC error to warning and disable the ORC related functionalities if ORC not found. 

Through that GraphAr can be built and ran without ORC support.

**Comment by yecol** [comment]
I see. I will take this issue and support this. 
Also I will revise the changes introduced by #552, to turn `termination` to warning.
